### PR TITLE
Extend available workspace params available to modify

### PR DESCRIPTION
**Pull request by mattseddon**
From https://discord.com/channels/485586884165107732/1177241185748779099

### Demo

https://github.com/iterative/vscode-dvc/assets/37993418/36a1369e-2062-4fd4-9962-f3b281a1a00b

https://github.com/iterative/vscode-dvc/assets/37993418/15eb884e-7b4a-4888-a0e7-a8347c2ed204

### CODE REVIEW

**Comment by dberenbaum** [comment]
Thanks @mattseddon! Have you tested if it works? I can't recall how vs code passes the param modifications to the experiment.

Also, not directly related, but we should not be showing `results/params.yaml` params in the list since they are not dependencies.

**Comment by mattseddon** [comment]
> Thanks @mattseddon! Have you tested if it works? I can't recall how vs code passes the param modifications to the experiment.

I have tested and it does seem to work. The command passed to the CLI is `dvc exp run -S params.yaml:model=small` which does seem to load the `small` config (I check `full` as well).

> Also, not directly related, but we should not be showing `results/params.yaml` params in the list since they are not dependencies.

All the code has ever done is show the full list of params. This is what we get from `exp show`:

```
"params": {
        "params.yaml": {
          "data": {
            "model": {
              "batch_size": 512,
              "latent_dim": 64,
              "duration": "00:00:30:00",
              "max_epochs": -1,
              "optim": { "lr": 0.01 }
            },
            "data_path": "fra.txt",
            "num_samples": 100000,
            "seed": 423
          }
        },
        "results/params.yaml": {
          "data": { "latent_dim": 8, "optim_params": { "lr": 0.01 } }
        }
      }
```

**Comment by dberenbaum** [comment]
> I have tested and it does seem to work. The command passed to the CLI is `dvc exp run -S params.yaml:model=small` which does seem to load the `small` config (I check `full` as well).

LGTM, thanks!

> All the code has ever done is show the full list of params. This is what we get from `exp show`:

Sorry, it's a repeat of https://github.com/iterative/vscode-dvc/issues/4088 and depends on https://github.com/iterative/dvc/issues/9588.

**Comment by dberenbaum** [comment]
Also a separate request, but can/should we hide `params.yaml:` when there is no other params file?

File: extension/src/experiments/model/modify/collect.ts [hunk 1]
```diff
@@ -8,32 +9,53 @@ export type Param = {
...
   }
+}
 
-  const path = appendColumnToPath(...pathArray)
+const removeFilePrefixIfOnlyDefault = (
+  acc: Param[],
+  paramsFiles: string[]
+): Param[] => {
+  const defaultParamsFile = 'params.yaml'
+  if (paramsFiles.length === 1 && paramsFiles[0] === defaultParamsFile) {
```
Comment (mattseddon) [inline extension/src/experiments/model/modify/collect.ts:39]: Addresses [this ask](https://github.com/iterative/vscode-dvc/pull/5030#issuecomment-1831910318).

**Review by sroy3 — APPROVED** [review]

**Review by julieg18 — APPROVED** [review]
Great work!

**Pull request merged**